subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -944,13 +949,17 @@ public TestFramework testFramework(@Nullable Closure testFrameworkConfigure) {
      *
      * @return The test framework options.
      */
-    @Internal
```

Comment:
This was an unnoticed regression - this needs to remain `@Nested` here.  For JUnit 4 alone - changing the `Category`s included or excluded should force the `Test` task to rerun and NOT result in an `UP-TO-DATE`.  This was only passing by accident.